api/v1alpha1/release_types_test.go

```diff
@@ -0,0 +1,1043 @@
+		// false
+		It("Condition is ConditionUnknown; should return false", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   releaseConditionType,
+							Status: metav1.ConditionUnknown,
+						},
+					},
+				},
+			}
+			Expect(r.HasSucceeded()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+
+		// condition.Status == metav1.ConditionTrue
+		// true
+		It("Condition is ConditionTrue; should return true", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   releaseConditionType,
+							Status: metav1.ConditionTrue,
+						},
+					},
+				},
+			}
+			Expect(r.HasSucceeded()).To(BeTrue(),
+				"Expected to get: %+v ; Actually Got: %+v\n", true, false)
+		})
+	})
+
+	// IsDone()
+	Describe("TestRelease_IsDone", func() {
+
+		// condition == nil
+		// false
+		It("condition is nil ; should return false", func() {
+			r := &Release{}
+			Expect(r.IsDone()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+
+		// condition.Status == metav1.ConditionTrue
+		// true
+		It("Condition is ConditionTrue ; should return true", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   releaseConditionType,
+							Status: metav1.ConditionTrue,
+						},
+					},
+				},
+			}
+			Expect(r.IsDone()).To(BeTrue(),
+				"Expected to get: %+v ; Actually Got: %+v\n", true, false)
+		})
+
+		// condition.Status == metav1.ConditionUnknown
+		// false
+		It("Condition is ConditionUnknown; should return false", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   releaseConditionType,
+							Status: metav1.ConditionUnknown,
+						},
+					},
+				},
+			}
+			Expect(r.IsDone()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+
+	})
+
+	// MarkFailed()
+	Describe("TestRelease_MarkFailed", func() {
+
+		// IsDone && Status.CompletionTime != nil
+		// no-op testcase
+		It("IsDone and Status.CompletionTime is not nil", func() {
+			args := commonArgs{
+				reason:  ReleaseReasonFoo,
+				message: "cow say m00",
+			}
+			r := &Release{
+				Status: ReleaseStatus{
+					StartTime: &metav1.Time{
+						Time: time.Now().Add(-time.Hour * 1),
+					},
+					CompletionTime: &metav1.Time{ // not nil
+						Time: time.Now(),
+					},
+					Conditions: []metav1.Condition{
+						{
+							Type:    releaseConditionType,
+							Status:  metav1.ConditionTrue,
+							Reason:  ReleaseReasonBar.String(),
+							Message: "Testcase one message string",
+						},
+					},
+				},
+			}
+
+			r.MarkFailed(args.reason, args.message)
+
+			Expect(len(r.Status.Conditions)).To(
+				Equal(1),
+				"Expected number of []Conditions to equal \"%+v\"\n",
+				1)
+
+			Expect(r.Status.Conditions[0].Status).To(
```

Comment:
You could use gomega matcher `MatchElements` to join all these `Expect`s

https://onsi.github.io/gomega/#testing-type-codestructcode

Reply:
That was a great suggestion! :+1: 
Thanks.
PTAL